### PR TITLE
Update starknet

### PR DESCRIPTION
**Pull request by michalsidzej**

### CODE REVIEW

File: packages/config/src/layer2s/starknet.ts [hunk 1]
```diff
@@ -166,26 +166,23 @@ export const starknet: Layer2 = {
       {
         name: 'WBTC Bridge',
         description: 'Starkgate bridge for WBTC.',
-        address: EthereumAddress('0x283751A21eafBFcD52297820D27C1f1963D9b5b4'),
-        upgradeability: discovery.getContract(
-          '0x283751A21eafBFcD52297820D27C1f1963D9b5b4',
-        ).upgradeability,
+        address: discovery.getContract('StarknetWBTCBridge').address,
```
Comment (antooni) [inline packages/config/src/layer2s/starknet.ts:169]: use ...discovery.getMainContractDetails()

File: packages/config/src/layer2s/starknet.ts [hunk 2]
```diff
@@ -218,15 +215,11 @@ export const starknet: Layer2 = {
...
-      name: 'Operator',
-      accounts: [
-        {
-          address: EthereumAddress(
-            '0x2C169DFe5fBbA12957Bdd0Ba47d9CEDbFE260CA7',
-          ),
-          type: 'EOA',
-        },
-      ],
+      name: 'Operators',
```
Comment (antooni) [inline packages/config/src/layer2s/starknet.ts:218]: use discovery.getPermissionedAccountsList(..)

**Review by antooni — CHANGES_REQUESTED** [review]

File: packages/backend/discovery/starknet/config.jsonc [hunk 3]
```diff
@@ -6,15 +6,31 @@
...
   ],
+  "names": {
+    "0x0437465dfb5B79726e35F08559B0cBea55bb585C": "DAI Bridge",
+    "0x283751A21eafBFcD52297820D27C1f1963D9b5b4": "WBTC Bridge",
+    "0xF6080D9fbEEbcd44D89aFfBFd42F098cbFf92816": "USDC Bridge",
+    "0xbb3400F107804DFB482565FF1Ec8D8aE66747605": "USDT Bridge"
+  },
   "overrides": {
+    // Starknet
+    // Make it possible to use the contract name without overriding it
```
Comment (antooni) [inline packages/backend/discovery/starknet/config.jsonc:20]: write `"0xc662c410C0ECf747543f5bA90660f6ABeBD9C8c4": "Starknet"` inside names for now

File: packages/config/src/layer2s/common/ProjectDiscovery.ts [hunk 4]
```diff
@@ -85,14 +85,11 @@ export class ProjectDiscovery {
...
-    assert(
-      isArray(value),
-      `Value of key ${key} does not exist in ${contractIdentifier} contract (${this.projectName})`,
-    )
+    assert(isArray(value), `Value of ${key} must be an array`)
 
     return value.map((account) => {
       assert(
-        isString(account) && new RegExp('^0x[a-fA-F\\d]{40}$').test(account),
+        isString(account) && EthereumAddress.test(account),
```
Comment (antooni) [inline packages/config/src/layer2s/common/ProjectDiscovery.ts:92]: use EthereumAddress.check

File: packages/shared/src/types/EthereumAddress.ts [hunk 5]
```diff
@@ -44,3 +44,8 @@ EthereumAddress.random = function random() {
 EthereumAddress.unsafe = function unsafe(address: string) {
   return address as unknown as EthereumAddress
 }
+
+const addressRegexp = new RegExp('^0x[a-fA-F\\d]{40}$')
```
Comment (antooni) [inline packages/shared/src/types/EthereumAddress.ts:48]: EthereumAddress.check gives this functionality

**Review by antooni — APPROVED** [review]
I proposed some small fixes, otherwise LGTM

File: packages/backend/discovery/starknet/config.jsonc (hunk 3 above)
Comment (michalsidzej) [reply packages/backend/discovery/starknet/config.jsonc:20]: I know it's possible. I just don't like it too much now

**Pull request merged**